polaris/evaluate/_metric.py

```diff
@@ -34,105 +137,229 @@ class MetricInfo(BaseModel):
-    cohen_kappa = MetricInfo(fn=cohen_kappa_score, direction="max")
-    pr_auc = MetricInfo(fn=average_precision_score, direction="max", y_type="y_score")
-
-    # binary only
-    f1 = MetricInfo(fn=f1_score, kwargs={"average": "binary"}, direction="max")
-    roc_auc = MetricInfo(fn=roc_auc_score, direction="max", y_type="y_score")
-
-    # multiclass tasks only
-    f1_macro = MetricInfo(fn=f1_score, kwargs={"average": "macro"}, direction="max")
-    f1_micro = MetricInfo(fn=f1_score, kwargs={"average": "micro"}, direction="max")
-    roc_auc_ovr = MetricInfo(
+DEFAULT_METRICS: dict[MetricLabel, MetricInfo] = {
+    "mean_absolute_error": MetricInfo(fn=mean_absolute_error, direction="min"),
+    "mean_squared_error": MetricInfo(fn=mean_squared_error, direction="min"),
+    "r2": MetricInfo(fn=r2_score, direction="max"),
+    "pearsonr": MetricInfo(fn=pearsonr, direction="max"),
+    "spearmanr": MetricInfo(fn=spearman, direction="max"),
+    "explained_var": MetricInfo(fn=explained_variance_score, direction="max"),
+    "absolute_average_fold_error": MetricInfo(fn=absolute_average_fold_error, direction=1),
+    "accuracy": MetricInfo(fn=accuracy_score, direction="max"),
+    "balanced_accuracy": MetricInfo(fn=balanced_accuracy_score, direction="max"),
+    "mcc": MetricInfo(fn=matthews_corrcoef, direction="max"),
+    "cohen_kappa": MetricInfo(fn=cohen_kappa_score, direction="max"),
+    "pr_auc": MetricInfo(fn=average_precision_score, direction="max", y_type="y_score"),
+    "f1": MetricInfo(fn=f1_score, kwargs={"average": "binary"}, direction="max"),
+    "roc_auc": MetricInfo(fn=roc_auc_score, direction="max", y_type="y_score"),
+    "f1_macro": MetricInfo(fn=f1_score, kwargs={"average": "macro"}, direction="max"),
+    "f1_micro": MetricInfo(fn=f1_score, kwargs={"average": "micro"}, direction="max"),
+    "roc_auc_ovr": MetricInfo(
         fn=roc_auc_score, kwargs={"multi_class": "ovr"}, direction="max", y_type="y_score"
-    )
-    roc_auc_ovo = MetricInfo(
+    ),
+    "roc_auc_ovo": MetricInfo(
         fn=roc_auc_score, kwargs={"multi_class": "ovo"}, direction="max", y_type="y_score"
-    )
-    # TODO: add metrics to handle multitask multiclass predictions.
+    ),
+    "rmsd_coverage": MetricInfo(fn=rmsd_coverage, direction="max", y_type="y_pred"),
+}
 
-    # docking related metrics
-    rmsd_coverage = MetricInfo(fn=rmsd_coverage, direction="max", y_type="y_pred")
 
-    @property
-    def fn(self) -> Callable:
-        """The callable that actually computes the metric"""
-        return self.value.fn
+class GroupedMetricConfig(BaseModel):
+    """
+    The configuration for a GroupedMetric
+
+    Attributes:
+        group_by: The column to group by.
+        on_error: How to handle errors when computing the metric.
+        default: The default value to use when an error occurs.
+        aggregation: The aggregation method to use when combining the metric scores.
+    """
+
```

Comment:
**Suggestion**
If the config is in this model, then it should own all the config.

```suggestion
    _kind: MetricKind = PrivateAttr("grouped")
```